ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/TransactionVerificationException.kt

```diff
@@ -0,0 +1,14 @@
+package net.corda.v5.ledger.common.transaction
+
+import net.corda.v5.application.flows.exceptions.FlowException
+import net.corda.v5.crypto.SecureHash
+
+/**
+ * Indicates that some aspect of the transaction named by [txId] violates the platform rules. The exact type of failure
+ * is expressed using a subclass. TransactionVerificationException is a [FlowException] and thus when thrown inside
+ * a flow, the details of the failure will be serialised, propagated to the peer and rethrown.
+ *
+ * @property txId the Merkle root hash (identifier) of the transaction that failed verification.
+ */
+abstract class TransactionVerificationException(val txId: SecureHash, message: String, cause: Throwable?)
```

Comment:
Do we want this to be abstract? I presume we are throwing some (undeclared) descendants of this, which doesn't seem very friendly. Think we should keep the number of specialized exceptions down to those which you might have an application reason to differentiate.

Reply:
The DP1 implementation has 21 subclasses of TransactionVerificationException.
Do we want to discard that hierarchy?
I thought with adding only their ancestor class, we keep the option to populate the childs as we go.

Reply:
I have changed this to abstract.